### PR TITLE
mpdecimal: fix compilation on ARM

### PR DESCRIPTION
**Pull request by neheb**
ARM 32-bit does not have -m32 on GCC. It's only needed on 64-bit
platforms anyway.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

### CODE REVIEW

File: subprojects/packagefiles/mpdecimal/meson.build [hunk 1]
```diff
@@ -3,7 +3,7 @@ project('mpdecimal', 'c', 'cpp',
         license: 'BSD-2-Clause',
         default_options: [
           'c_std=c99',
-          'cpp_std=c++11',
+          'cpp_std=c++14',
```
Comment (dnicolodi) [inline subprojects/packagefiles/mpdecimal/meson.build:6]: This hunk seems unrelated, or is c++14 required on ARM for some reason?

Comment (neheb) [reply subprojects/packagefiles/mpdecimal/meson.build:6]: https://github.com/mesonbuild/wrapdb/actions/runs/2607518219

Needed for MSVC.

Comment (dnicolodi) [reply subprojects/packagefiles/mpdecimal/meson.build:6]: I am sure that the CI compiled mpdecimal with `cpp_std=c++11` before. Did something change in Meson to make it fail when this setting is used with MSVC?

Comment (neheb) [reply subprojects/packagefiles/mpdecimal/meson.build:6]: Yes.

subprojects _std settings recently started being respected. Before, they would inherit the _std settings of the host project which in this case happens to be c++17: https://github.com/mesonbuild/wrapdb/blob/master/meson.build

**Comment by neheb** [comment]
This is ready to go.

File: subprojects/packagefiles/mpdecimal/meson.build (hunk 1 above)
Comment (dcbaker) [reply subprojects/packagefiles/mpdecimal/meson.build:6]: That warning should be non-fatal, this is a Meson bug

**Comment by neheb** [comment]
ping @jpakkane

File: subprojects/packagefiles/mpdecimal/meson.build (hunk 1 above)
Comment (skrah) [reply subprojects/packagefiles/mpdecimal/meson.build:6]: The way in which the `mpdecimal` build machinery deals with this is to set `--std=c++11` for everything except `MSVC`.  `MSVC` automatically selects the lowest supported C++ version (currently `C++14`) if no standard is given.

`meson` could just omit the standard for `MSVC` if the selected standard is `<= C++14`.
